extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -47,27 +51,33 @@ private void doHandle(RoutingContext ctx, ManagedContext requestContext) {
         }
         SmallRyeHealthReporter reporter = Arc.container().instance(SmallRyeHealthReporter.class).get();
         Context context = Vertx.currentContext();
-        getHealth(reporter, ctx).emitOn(MutinyHelper.executor(context))
-                .subscribe().with(health -> {
-                    if (requestContext != null) {
-                        requestContext.terminate();
-                    }
-                    HttpServerResponse resp = ctx.response();
-                    if (health.isDown()) {
-                        resp.setStatusCode(503);
-                    }
-                    resp.headers().set(HttpHeaders.CONTENT_TYPE, "application/json; charset=UTF-8");
-                    Buffer buffer = Buffer.buffer(256); // this size seems to cover the basic health checks
-                    try (BufferOutputStream outputStream = new BufferOutputStream(buffer);) {
-                        reporter.reportHealth(outputStream, health);
-                        resp.end(buffer);
-                    } catch (IOException e) {
-                        throw new UncheckedIOException(e);
-                    }
-                }, failure -> {
-                    if (requestContext != null) {
-                        requestContext.terminate();
-                    }
-                });
+        Uni<SmallRyeHealth> healthUni = getHealth(reporter, ctx);
+        if (context != null) {
+            healthUni = healthUni.emitOn(MutinyHelper.executor(context));
+        } else {
+            // No context provided, fallback to the blocking execution
+            LOG.warn("Vert.x context is unavailable in the health check invocation. Falling back to the blocking execution.");
```

Comment:
I don't think this warning is useful. If `Vertx.currentContext()` is `null`, clearly you're not on a Vert.x event loop thread (because you're not on a Vert.x thread), so you can assume blocking is fine IMHO.

I'd lower this to debug.

Reply:
The rationale is that it is generally because of one of the previous routes (like authentication) that does not switch on a Vert.x thread.

Reply:
@Ladicek check the linked PR discussion please for context.

Reply:
Why should _health_, of all places, warn the user that they got threading wrong? If there's a custom auth mechanism that offloads work to a custom thread, either it's OK and health shouldn't warn about it, or it's not OK and _something else than health_ should warn about it, because it's not just health that's affected.

Reply:
Because health is pulling up Vert.x context and relies on it to be present? If you check other uses of Vertx.currentContext() we're doing this all over the place around Quarkus.

Reply:
Well, with this PR, we are no longer _relying_ on it.

Reply:
I mean, you do you. I'm no expert on health and this PR shouldn't be blocked by my lack of understanding.

Reply:
Yes, actually, while I was suggesting the warning, I believe with this PR it can be omitted. 
I think the warning should go in quarkus-security where things tend to be wrong (it's not the first report where I see an authenticator emitting on a random thread).